File: new/has_duplicates_hash_set_length.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // create an unordered_set for the nums with nums.begin() -> first element
        // to the one after last element nums.end() -> points one next to last element
        // get the size of the set, it should have size of set with no duplicates. 
        // if that size is less than size of list, then we have duplicate. 
        return unordered_set<int>(nums.begin(), nums.end()).size() < nums.size();
        // time: O(n) for checking if each item already exists in unordered set, if not creating and putting it in hash map. 
        // space: O(n) for internal hashmap by undordered_set
    }
};

File: new/python/has_duplicates_hash_set_length.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # create a set from the list, and compare the lengths of set to list
        # if the set's length is less than list's length, then we have duplicates
        return len(set(nums)) < len(nums)
        # time complexity: O(n) b/c set(nums) internally creates hash map, and
        # checks if each element in the list is already there. Each operation is
        # O(1) and there are n elements in nums. O(n).
        # Space complexity: O(n), for the hash map
